workspaces/bulk-import/plugins/bulk-import-backend/src/service/handlers/import/bulkImports.ts

```diff
@@ -156,8 +165,9 @@ export async function findAllImports(
     });
 
   // sorting the output to make it deterministic and easy to navigate in the UI
-  sortImports(imports);
-
+  if (sortColumn) {
+    sortImports(imports, sortColumn, sortOrder);
+  }
```

Comment:
Are we always fetching ALL elements in the backend?

And guess that we are and should not do that. So this sort function will only sort the current window (fetched page).

Instead of that we should pass the sorting to the backend calls like GitHub and the catalog API and remove the sorting from our backend.

Reply:
Yes, it would be the best if we could sort and paginate on the Github side.
I believe we are at the backend unfortunately fetching for `imports` all Catalog locations and then filtering those accessible for user via integrations. @rm3l would be the best to answer if this could be adjusted to be more efficient. Maybe paginating and sorting can be added to [filterLocationsAccessibleFromIntegrations](https://github.com/redhat-developer/rhdh-plugins/blob/main/workspaces/bulk-import/plugins/bulk-import-backend/src/github/githubApiService.ts#L372-#L417).

Reply:
Yes, it would be the best if we could sort and paginate at the Github side. Unfortunately, for `/imports` endpoint, I believe we are fetching all Catalog locations and then filtering those reachable with GH integrations. @rm3l might have valuable insights into the reasoning behind this solution and whether we can approach this more effectively. Maybe we can adjust [filterLocationsAccessibleFromIntegrations](https://github.com/redhat-developer/rhdh-plugins/blob/main/workspaces/bulk-import/plugins/bulk-import-backend/src/service/handlers/import/bulkImports.ts#L116-#L120) [here](https://github.com/redhat-developer/rhdh-plugins/blob/main/workspaces/bulk-import/plugins/bulk-import-backend/src/github/githubApiService.ts#L372-#L430).

When [fetching Catalog locations](https://github.com/redhat-developer/rhdh-plugins/blob/main/workspaces/bulk-import/plugins/bulk-import-backend/src/service/handlers/import/bulkImports.ts#L90-#L96), `pageNumber` and `pageSize` are passed as params but not really used there, since we don't know which ones are actually reachable GH integrations. Maybe the best approach would be to switch stuff around and first fetch GH integrations, maybe it would help performance.